system/web/services/HandlerService.cfc

```diff
@@ -689,17 +692,47 @@ component extends="coldbox.system.web.services.BaseService" accessors="true" {
 							""
 						);
 						mdEntry.provider = arguments.ehBean.getActionMetadata( "cacheProvider", "template" );
+                        mdEntry.cacheInclude = arguments.ehBean.getActionMetadata( "cacheInclude", "*" );
+                        mdEntry.cacheExclude = arguments.ehBean.getActionMetadata( "cacheExclude", "" );
+                        mdEntry.cacheFilter = arguments.ehBean.getActionMetadata( "cacheFilter", "" );
 
 						// Handler Event Cache Key Suffix, this is global to the event
 						if (
-							isClosure( arguments.oEventHandler.EVENT_CACHE_SUFFIX ) || isCustomFunction(
-								arguments.oEventHandler.EVENT_CACHE_SUFFIX
-							)
+							isClosure( arguments.oEventHandler.EVENT_CACHE_SUFFIX ) || 
+                            isCustomFunction( arguments.oEventHandler.EVENT_CACHE_SUFFIX )
 						) {
 							mdEntry.suffix = oEventHandler.EVENT_CACHE_SUFFIX( arguments.ehBean );
 						} else {
 							mdEntry.suffix = arguments.oEventHandler.EVENT_CACHE_SUFFIX;
 						}
+
+                        // if the cacheFilter has a length and is a method, then we need to verify and store the resulting closure
+                        if ( len( mdEntry.cacheFilter ) ) {
```

Comment:
This won't work on a closure/lambda.  Just do if `!isSimpleValue( mdEntry.cacheFilter )`